docs/reference/configuration.md

```diff
@@ -0,0 +1,210 @@
+|         | wal_dir  | String | Write-ahead log directory, "/tmp/greptimedb/wal"    |
+| storage |          |        | Storage engine options                              |
+|         | type     | String | Storage engine type, Only supports 'File' right now |
+|         | data_dir | String | Data directory, "/tmp/greptimedb/data" by default   |
+
+
+
+## Standalone
+
+When you use GreptimeDB in the standalone mode, you can configure it as below:
+
+```toml
+node_id = 0
+mode = 'standalone'
+http_addr = '127.0.0.1:4000'
+datanode_mysql_addr = "127.0.0.1:3306"
+datanode_mysql_runtime_size = 4
+wal_dir = "/tmp/greptimedb/wal/"
+
+[storage]
+type = 'File'
+data_dir = '/tmp/greptimedb/data/'
+
+[grpc_options]
+addr = '127.0.0.1:4001'
+runtime_size = 8
+
+[mysql_options]
+addr = '127.0.0.1:4002'
+runtime_size = 2
+
+[influxdb_options]
+enable = true
+
+[opentsdb_options]
+addr = "127.0.0.1:4242"
+enable = true
+runtime_size = 2
+
+[prometheus_options]
+enable = true
+
+[postgres_options]
+addr = '127.0.0.1:4003'
+runtime_size = 2
+check_pwd = false
+```
+
+Some specific options for standalone:
+
+* `datanode_mysql_addr` and `datanode_mysql_runtime_size`, to set datanode's MySQL server configurations.
+
+## Frontend in distributed mode
+
+Configure frontend in distributed mode:
+
+```toml
+mode = "distributed"
+datanode_rpc_addr = '127.0.0.1:3001'
+http_addr = '127.0.0.1:4000'
+
+[meta_client_opts]
+metasrv_addr = "1.1.1.1:3002"
+timeout_millis = 3000
+connect_timeout_millis = 5000
+tcp_nodelay = false
+```
+
+The `datanode_rpc_addr` is not used in `distributed` mode, you can leave it in default value.
+
+The `meta_client_opts` configure the metasrv client confugrations, incuding:
+
+* `metasrv_addr`, metasrv address.
+* `timeout_millis`, operation timeout in milliseconds, 3000 by default.
+* `connect_timeout_millis`, connect server timeout in milliseconds,5000 by default.
+
+
+## Datanode in distributed mode
+
+Configure datanode in distributed mode:
+
+```toml
+node_id = 42
+mode = "distributed"
+rpc_addr = '127.0.0.1:3001'
+wal_dir = '/tmp/greptimedb/wal'
+rpc_runtime_size = 8
+mysql_addr = '127.0.0.1:3306'
+mysql_runtime_size = 4
+
+[storage]
+type = 'File'
+data_dir = '/tmp/greptimedb/data/'
+
+[meta_client_opts]
+metasrv_addr = "1.1.1.1:3002"
+timeout_millis = 3000
+connect_timeout_millis = 5000
+tcp_nodelay = false
+```
+
+Datanode in distributed mode should set different `node_id` in different nodes.
+
+## Metasrv configuration
+
+A sample configurations:
+
+```toml
+bind_addr = '127.0.0.1:3002'
+server_addr = '127.0.0.1:3002'
+store_addr = '127.0.0.1:2379'
+datanode_lease_secs = 30
+```
+
+| Key                 | Type    | Description                                                                                                             |   |
+|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|---|
```

Comment:
```suggestion
|---------------------|---------|-------------------------------------------------------------------------------------------------------------------------|
```

Looks like there is an empty column in the end. Other rows should be change correspondingly.